.github/workflows/connector-test-command.yml

```diff
@@ -202,84 +232,82 @@ jobs:
             --print-ci-secrets-masks
 
       - name: Run Unit Tests
-        if: ${{ steps.get-language.outputs.LANGUAGE }}
+        if: ${{ matrix.connector && steps.get-language.outputs.LANGUAGE }}
         run: |
           poe connector ${{ matrix.connector }} test-unit-tests
 
       - name: Run Integration Tests
-        if: ${{ steps.get-language.outputs.LANGUAGE }}
+        if: ${{ matrix.connector && steps.get-language.outputs.LANGUAGE }}
         run: |
           poe connector ${{ matrix.connector }} test-integration-tests
 
       - name: Publish Test Results
         uses: EnricoMi/publish-unit-test-result-action@v2
-        if: always()
+        if: always() && matrix.connector
         with:
           check_name: "`${{ matrix.connector }}` Connector Test Results"
           files: |
             airbyte-integrations/connectors/${{ matrix.connector }}/build/test-results/**/*.xml
 
-  test-summary:
-    name: Connector Test Summary Check
-    needs: [connectors-test-matrix]
-    runs-on: ubuntu-24.04
-    steps:
-      - name: Report Test Summary
-        run: echo "✅ All connector tests completed successfully."
-
-  test-summary-noop:
-    name: Connector Test Summary Check
-    needs: [generate-matrix]
-    if: ${{ ! needs.generate-matrix.outputs.connector-matrix }}
+  non-jvm-connectors-test-summary:
+    name: Non-JVM Connector Test Summary Check
+    needs: [non-jvm-connectors-test-matrix]
     runs-on: ubuntu-24.04
     steps:
       - name: Report Test Summary
-        run: echo "🟦 No connectors updated. All good here."
+        run: echo "✅ All non-JVM connector tests completed successfully (or nothing to test)."
```

Comment:
Status message has been updated to reflect that sometimes this will also be reached if no connectors change.